### PR TITLE
CNS-586: TX claim rewards

### PR DESCRIPTION
**Pull request by oren-lava**
implemented `tx dualstaking claim-rewards [optional: provider] --from <delegator>`
The TX claims all rewards by default. By specfiying a provider, the delegator can claim from a specific provider only

### CODE REVIEW

File: x/dualstaking/keeper/delegator_reward.go [hunk 1]
```diff
@@ -92,3 +94,43 @@ func (k Keeper) CalcDelegationsSum(stakeEntry epochstoragetypes.StakeEntry) math
...
+	if err != nil {
+		return utils.LavaFormatWarning("could not claim delegator rewards", err,
+			utils.Attribute{Key: "delegator", Value: delegator},
+		)
+	}
+
+	for _, reward := range res.Rewards {
+		delegatorAcc, err := sdk.AccAddressFromBech32(delegator)
+		if err != nil {
+			utils.LavaFormatError("could not claim delegator reward from provider", err,
```
Comment (orenl-lava) [inline x/dualstaking/keeper/delegator_reward.go:110]: this is really bad, should be "critical"

File: x/pairing/keeper/delegator_rewards_test.go [hunk 2]
```diff
@@ -248,12 +261,32 @@ func TestProviderRewardWithCommission(t *testing.T) {
...
+	reward, found := ts.Keepers.Dualstaking.GetDelegatorReward(ts.Ctx, ind)
+	require.True(ts.T, found)
+
+	_, err := ts.Servers.DualstakingServer.ClaimRewards(ts.Ctx, dualstakingtypes.NewMsgClaimRewards(delegator.String(), provider))
+	require.Nil(ts.T, err)
+
+	newBalance := ts.GetBalance(delegator)
+	require.Equal(ts.T, balance+reward.Amount.Amount.Int64(), newBalance)
+
+	_, found = ts.Keepers.Dualstaking.GetDelegatorReward(ts.Ctx, ind)
```
Comment (orenl-lava) [inline x/pairing/keeper/delegator_rewards_test.go:288]: maybe also (or instead) use the query again to verify that nothing is left?

**Review by orenl-lava — CHANGES_REQUESTED** [review]

File: x/pairing/keeper/delegator_rewards_test.go (hunk 2 above)
Comment (oren-lava) [reply x/pairing/keeper/delegator_rewards_test.go]: done. see https://github.com/lavanet/lava/commit/302b2bec531bd9308d142d746c327509bde9b547

File: x/dualstaking/keeper/delegator_reward.go (hunk 1 above)
Comment (oren-lava) [reply x/dualstaking/keeper/delegator_reward.go]: done. see https://github.com/lavanet/lava/commit/302b2bec531bd9308d142d746c327509bde9b547

**Review by orenl-lava — APPROVED** [review]

**Pull request merged**